app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  def new
+    @comments = Comment.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = @user.posts.find(params['post_id'])
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to user_post_path(@user, @post) }
+      else
+        format.html { render :new }
+      end
+    end
+  end
```

Comment:
- [ ] Comments are not created correctly. This is as a result of not passing the parameter to view correctly.
I looked into your code to find the cause of this problem and I noticed that in your `comments#new` you are passing `@comments` to view instead of `@comment`

Please do this instead 👇 
```ruby
def new
  @comment = Comment.new
end
```

- [ ] You are also finding the `post` twice. On line 8 and again on line 11. The latter is no longer necessary since you now have the `@post` variable on line 8